kubernetes/kubernetes.go

```diff
@@ -413,6 +440,65 @@ func (ka KubeAgentConfig) sendMetrics(metricSample *os.File) {
 	}
 }
 
+func (ka KubeAgentConfig) sendMetricsBasedOnUploadMode(customS3Mode bool, metricSample *os.File) {
+	if customS3Mode {
+		log.Infof("Uploading Metrics to Custom S3 Bucket %s", ka.CustomS3UploadBucket)
+		go ka.sendMetricsToCustomS3(metricSample)
+	} else {
+		log.Info("Uploading Metrics")
+		go ka.sendMetrics(metricSample)
+	}
+}
+
+func (ka KubeAgentConfig) sendMetricsToCustomS3(metricSample *os.File) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(ka.CustomS3Region)},
```

Comment:
Would consider using `WithMaxRetries` retries here to ensure the agent doesn't crash over misc network issues.

Reply:
added